### ISSUE TITLE
W1b: a non-rigid augmentation in `AugmentationSequential` silently desynchronizes the data keys

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4420]
`conventions.rst` states that one `AugmentationSequential` call applies the same sampled transform to every registered data type. That holds for rigid augmentations, which propagate through a transformation matrix: `RandomAffine` moves the image, the keypoints, the boxes and the mask together, and `.inverse()` restores the keypoints exactly.

A non-rigid augmentation has no matrix, and the container has no fallback for it. `RandomElasticTransform` warps the image and returns the keypoints and the boxes byte-identical to what was passed in; its mask **is** warped, through its own `apply_transform_mask` (the same displacement field, nearest resampling) — see the correction below the Observed block. `RandomThinPlateSpline` and `RandomFisheye` return keypoints and boxes unchanged in the same way, but raise a bare `NotImplementedError` — the exception carries no message at all — as soon as a `MASK` key is registered, so the three non-rigid classes do not even agree with each other about which keys are silently passed through and which refuse.

For a segmentation or detection pipeline this is the failure mode that cannot be detected downstream: the image has moved and its labels have not, training proceeds, and nothing raises. The machinery already knows the operation is unsupported in a related place — `transformation_matrix_mode="rigid"` raises when a non-rigid op is in the chain and the caller asks for the matrix — so the gap is on the data-key path only.

## Repro

```python
import torch
import kornia.augmentation as K
from kornia.augmentation import AugmentationSequential

torch.manual_seed(42)
x = torch.rand(2, 3, 6, 8)
kp = torch.tensor([[[1.0, 2.0], [5.0, 4.0]]]).repeat(2, 1, 1)
box = torch.tensor([[[0.0, 0.0, 3.0, 2.0]]]).repeat(2, 1, 1)
m = torch.zeros(2, 1, 6, 8)
m[:, :, 1:4, 2:6] = 1.0

def move(build, key, payload):
    torch.manual_seed(0)
    seq = AugmentationSequential(build(), data_keys=["input", key])
    out_img, out_key = seq(x, payload)
    return (out_img - x).abs().max().item(), (out_key - payload).abs().max().item()

CASES = [
    ("RandomElasticTransform", lambda: K.RandomElasticTransform(p=1.0)),
    ("RandomThinPlateSpline",  lambda: K.RandomThinPlateSpline(p=1.0)),
    ("RandomFisheye",          lambda: K.RandomFisheye(torch.tensor([-0.3, 0.3]), torch.tensor([-0.3, 0.3]),
                                                       torch.tensor([0.9, 1.0]), p=1.0)),
    ("RandomAffine (rigid control)", lambda: K.RandomAffine(degrees=(45.0, 45.0), p=1.0)),
]
for name, build in CASES:
    print(name)
    for key, payload in (("keypoints", kp), ("bbox_xyxy", box), ("mask", m)):
        try:
            img_mv, key_mv = move(build, key, payload)
            print(f"    {key}=(img {img_mv:.6f}, {key} {key_mv:.6f})")
        except Exception as e:
            print(f"    {key}={type(e).__name__}({e!r})")
```

## Observed

```
RandomElasticTransform
    keypoints=(img 0.717043, keypoints 0.000000)
    bbox_xyxy=(img 0.717043, bbox_xyxy 0.000000)
    mask=(img 0.717043, mask 0.000000)
RandomThinPlateSpline
    keypoints=(img 0.886642, keypoints 0.000000)
    bbox_xyxy=(img 0.886642, bbox_xyxy 0.000000)
    mask=NotImplementedError(NotImplementedError())
RandomFisheye
    keypoints=(img 0.999804, keypoints 0.000000)
    bbox_xyxy=(img 0.999804, bbox_xyxy 0.000000)
    mask=NotImplementedError(NotImplementedError())
RandomAffine (rigid control)
    keypoints=(img 0.920752, keypoints 1.621320)
    bbox_xyxy=(img 0.920752, bbox_xyxy 1.914213)
    mask=(img 0.920752, mask 1.000000)
```

**Correction (2026-09-11, found by the batch's own re-review):** the `mask 0.000000` row for `RandomElasticTransform` is a fixture artifact, not a fact. The class defines `apply_transform_mask`, which warps the mask with the same displacement field as the image using nearest resampling; the repro's uniform block mask at the default `alpha=(1.0, 1.0)` happens to map every pixel back onto a pixel of the same value. Executed at `95294372` (torch 2.14.0, cpu, `(1, 1, 6, 8)`, seed 0, inside `AugmentationSequential(..., data_keys=["input", "mask", "keypoints"])`): a checkerboard mask changes 3/48 pixels at `alpha=1.0` and 18/48 at `alpha=5.0`; the block mask changes 6/48 at `alpha=5.0`; the keypoints move by `0.0` in every case. So for `RandomElasticTransform` the desynchronization is **keypoints and boxes only**; for `RandomThinPlateSpline` / `RandomFisheye` it is keypoints and boxes plus the bare `NotImplementedError` on a mask. The classification and disposition below are unchanged; the pin and the docstrings of batch 6a record the corrected split.

The `0.0` keypoint / box move is stable one parameter away from the fixture: for `RandomElasticTransform` the image/keypoint move is `(0.741906, 0.0)` at `B=1`, `(0.744145, 0.0)` at `B=2` and `(0.745797, 0.0)` at `B=4, same_on_batch=True` (the third element of the original triple, the mask, was the artifact corrected above).

## Expected

Two coherent outcomes, and both change results for existing pipelines:

1. **Warp the other data keys.** A dense displacement field can be sampled at keypoint and box-corner positions, as `RandomElasticTransform` already does for its mask with nearest resampling, so the container propagates the same warp to every registered key, as it does for a matrix.
2. **Refuse.** Registering a mask, box or keypoint key alongside a non-rigid augmentation raises a kornia message naming the class and the key, the way `transformation_matrix_mode="rigid"` already refuses the matrix.

Whichever is chosen, the three non-rigid classes must agree, and the bare `NotImplementedError` on the `MASK` path of `RandomThinPlateSpline` and `RandomFisheye` needs a message either way — an exception with no argument gives a caller nothing to act on.

## Classification

**Broken output** — the image is warped away from its own mask, boxes and keypoints, silently, in the one container whose stated purpose is keeping them in step. Idiom: 4 (a degenerate case the machinery cannot handle produces garbage instead of raising).

## Disposition

`MERGE IN WINDOW — behavior change`. Either outcome changes what existing pipelines get from a call that currently succeeds, so this is the behavior-change discussion CONTRIBUTING asks for: **in the coordinated Repair & Deprecation window, not before.** Batch 6 documents the current pass-through on `conventions.rst` and on the three classes, and pins it.

## Environment

kornia 0.9.0rc1 @ `d6a2c138` (main), torch 2.14.0, python 3.11, cpu. Found during the augmentation conventions audit (batch 6, tracking issue #4407); the batch documents the behavior above as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

**Comment by ducha-aiki** [comment]
Correction to the body above, from the batch's own re-review of the 6a docstrings: `RandomElasticTransform` **does** warp the mask. It defines `apply_transform_mask`, which applies the same displacement field with nearest resampling, and the original repro's uniform block mask at the default `alpha=(1.0, 1.0)` simply mapped every pixel onto a pixel of the same value, so the `mask 0.000000` row was a symmetric-fixture artifact. Executed at `95294372` (torch 2.14.0, cpu): a checkerboard `(1, 1, 6, 8)` mask changes 3/48 pixels at `alpha=1.0` and 18/48 at `alpha=5.0`, while the keypoints move by `0.0` in every case.

The issue stands as filed for keypoints and boxes (all three classes) and for the bare `NotImplementedError` on a mask (`RandomThinPlateSpline`, `RandomFisheye`); the body has been patched in place to say so, and the batch-6a pin and docstrings record the corrected split. Classification and disposition are unchanged.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
